staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3102,6 +3102,10 @@ type PodSpec struct {
 	// Restart policy for all containers within the pod.
 	// One of Always, OnFailure, Never.
 	// Default to Always.
+	// When the RestartPolicy field is specified in the Pod Template
+	// for Deployment/ReplicaSet/DaemonSet/StatefulSet,
+	// only a .spec.template.spec.restartPolicy equal to Always is allowed,
+	// which is the default if not specified.
```

Comment:
documentation about use inside other workload objects probably isn't appropriate to put on the generic PodSpec type

Reply:
So what are some good suggestions to avoid user misunderstanding?

Reply:
we could make a generic statement about some uses of pod spec restricting the value further, but this shouldn't enumerate those uses

Reply:
> we could make a generic statement about some uses of pod spec restricting the value further.

Do you mean to restrict here or in other places?  
https://github.com/kubernetes/kubernetes/blob/619f1cf55208212893504338f18c69f3dcfcdc66/staging/src/k8s.io/api/core/v1/types.go#L3063

Reply:
On the RestartPolicy field inside PodSpec, we could add a line like "In some contexts, only a subset of those values may be permitted."

Then in the StatefulSetSpec, DaemonSetSpec, ReplicaSetSpec, ReplicationControllerSpec, JobSpec, CronJobSpec types, on the Template field, we could add a line to the godoc indicating the additional restrictions on the restart policy for the template.